CMakeLists.txt

```diff
@@ -737,7 +737,12 @@ else()
     # PIE executables randomly crash at startup with ASAN
     # Windows binaries die on startup with PIE when compiled with GCC <9.x
     # Windows dynamically-linked binaries die on startup with PIE regardless of GCC version
-    add_linker_flag_if_supported(-pie LD_SECURITY_FLAGS)
+    if (APPLE)
+      # Clang does not support -pie flag
+      add_linker_flag_if_supported("-Wl,-pie" LD_SECURITY_FLAGS)
```

Comment:
I don't understand. `-pie` doesn't work, but `-Wl,-pie` does?

Reply:
Clang does not support -pie, the linker itself does support it.

```
       -Wl,<args>
              Pass the comma separated arguments in args to the linker.
```

Updated the PR to test for Clang and not for APPLE.